src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,270 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    private static final String TIME_INPUT_PATTERN = "dd-MM-yyyy HHmm";
+    private static final String DEADLINE_TIME_INPUT_PATTERN = "dd-MM-yyyy";
```

Comment:
Good code quality practice as to avoid magic numbers. I like how you always keep things clear and simple.